### PR TITLE
fix #4938: i18n doesn't handle node.title correctly

### LINKED ISSUE #4938: i18n doesn't handle node.title correctly; contents, topic, admonition, table and section
Original report is: https://groups.google.com/d/msg/sphinx-users/hdzuCi7AvRY/etw-3qWBBAAJ

### Problem
There are two problems

1. `make gettext` doesn't generate msgid for title of `topic` directive and `contents` directive

ex.
```
.. contents:: Contents Title

.. topic:: Topic Title
```

2. `make html` with i18n catalog doesn't apply translated message if msgid starts with `1. `

ex. `index.rst`
```
1. A Header
===========

.. admonition:: 1. Admonition Title

   Admonition body

.. topic:: 1. Topic Title

   Topic body

.. contents:: 1. Contents Title

.. table:: 1. Table Title

   +---+
   |Foo|
   +---+
```

#### Procedure to reproduce the problem
`conf.py`:
```
locale_dirs = ['./locale']
language = 'ru'
gettext_compact = False
```

commands:
```
make gettext
sphinx-intl update -l ru
# edit ./locale/ru/LC_MESSAGES/index.po
make html
```

#### Error logs / results
no errors

#### Expected results
All text in the `index.rst` are translated with using `index.po` catalog.

### Reproducible project / your project
[4938.zip](https://github.com/sphinx-doc/sphinx/files/1977426/4938.zip)

### Environment info
- OS: Any (Windows 10 for me)
- Python version: 3.6.1
- Sphinx version: 1.7.4

### PR DESCRIPTION
**Pull request by shimizukawa**
### Feature or Bugfix
- Bugfix for #4938 

### Purpose
1. `make gettext` generates msgid for title of `topic` directive and `contents` directive
2. `make html` with i18n catalog apply translated message even if msgid starts with `1. `

### Detail
see #4938

### CODE REVIEW

File: sphinx/transforms/i18n.py
Comment (shimizukawa) [inline sphinx/transforms/i18n.py:255]: I don't think this code is good. I should refactor this `sphinx.transform.i18n` module later.

File: sphinx/util/nodes.py
Comment (shimizukawa) [inline sphinx/util/nodes.py:53]: this comment lines will uncomment after merging master

File: sphinx/util/nodes.py
Comment (tk0miya) [inline sphinx/util/nodes.py]: What is difference from `translatable` nodes?

On i18n transform, it is described as following:
```
# ignore block markups
if not isinstance(patch, ALLOWED_TRANSLATED_NODES):
    continue  # skip
```
Are these same meaning?

**Review by tk0miya — APPROVED** [review]
LGTM with nits

File: sphinx/util/nodes.py
Comment (shimizukawa) [reply sphinx/util/nodes.py]: Not same.
When `msgstr` in po files contain block markup, `if not isinstance(patch, ALLOWED_TRANSLATED_NODES):` ignore such translations.

Comment (tk0miya) [reply sphinx/util/nodes.py]: I feel it is confusable. Could you rename this please to more appropriate name.

Comment (shimizukawa) [reply sphinx/util/nodes.py]: I thought about defining it by naming rather than listing node classes in the code of `i18n.py`. So, I named `ALLOWED_TRANSLATED_NODES` to them, but this was not a property of being given a single name.
The nodes.title that added this time is a hacky code to handle in pair with the processing in `i18n.py`.
I'll revert this part and add a comment for each node classes.

**Comment by shimizukawa** [comment]
I'll merge this after test passing

**Pull request merged**
